oauth2/src/main/java/com/okta/spring/boot/oauth/OktaOAuth2Configurer.java

```diff
@@ -62,21 +66,49 @@ public void init(HttpSecurity http) throws Exception {
+                    return;
+                }
 
-            // resource server configuration
-            if (!context.getBeansOfType(OAuth2ResourceServerProperties.class).isEmpty()) {
-                OAuth2ResourceServerProperties resourceServerProperties = context.getBean(OAuth2ResourceServerProperties.class);
-                if (!isEmpty(resourceServerProperties.getJwt().getIssuerUri())) {
-                    // configure Okta specific auth converter (extracts authorities from `groupsClaim`
-                    configureResourceServer(http, oktaOAuth2Properties);
-                } else {
-                    log.debug("OAuth resource server not configured due to missing issuer property");
+                OAuth2ResourceServerConfigurer oAuth2ResourceServerConfigurer =
+                    http.getConfigurer(OAuth2ResourceServerConfigurer.class);
+
+                if (oAuth2ResourceServerConfigurer != null) {
+
+                    // code below is wrapped in AccessController to address findbugs error 'DP_DO_INSIDE_DO_PRIVILEGED'
+                    Field jwtConfigurerField = (Field) AccessController.doPrivileged((PrivilegedAction) () -> {
+                        Field result = null;
+                        try {
+                            result = OAuth2ResourceServerConfigurer.class.getDeclaredField("jwtConfigurer");
+                            result.setAccessible(true);
+                        } catch (NoSuchFieldException e) {
+                            log.warn("Could not get field 'jwtConfigurer' of {} via reflection",
+                                OAuth2ResourceServerConfigurer.class.getName(), e);
+                        }
+                        return result;
+                    });
+
+                    if (jwtConfigurerField != null) {
+                        OAuth2ResourceServerConfigurer.JwtConfigurer jwtConfigurerValue =
+                            (OAuth2ResourceServerConfigurer.JwtConfigurer) jwtConfigurerField.get(oAuth2ResourceServerConfigurer);
+
+                        if (jwtConfigurerValue != null) {
+                            log.info("JWT configurer is set in OAuth resource server configuration. " +
+                                "JWT validation will be configured.");
+                            http.oauth2ResourceServer()
+                                .jwt().jwtAuthenticationConverter(new OktaJwtAuthenticationConverter(oktaOAuth2Properties.getGroupsClaim()));
+                        } else {
+                            log.info("JWT configurer is NOT set in OAuth resource server configuration. " +
+                                "Opaque Token validation/introspection will be configured.");
+                            configureResourceServerForOpaqueTokenValidation(http, oktaOAuth2Properties);
+                        }
+                    } else {
+                        log.warn("JWT configurer field was not found in OAuth resource server configuration");
```

Comment:
nit: Add something about checking this project for updates, https://github.com/okta/okta-spring-boot

Thinking about this again 🤔 we should probably fail in this case (still with a note about checking the project page)

My thoughts around this are our JWT checking validates more than the out of the box Spring Sec config.
Defaulting to a weaker security stance could raise other issues.  So just fail, with a nice long error message about the compatibility between versions, and checking for updates on the GitHub project's page.

A user should ONLY see this type of message when updating one of the versions without the other, so failing fast  is probably the best bet

(sorry for going back and forth on this)

Reply:
will update.